File: project files/server/src/models/ParentOrder.js
const mongoose = require('mongoose');

const childSummarySchema = new mongoose.Schema(
  {
    orderId: { type: String, required: true },
    sellerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    orderStatus: { type: String, required: true },
    paymentStatus: { type: String, required: true },
    totalAmount: { type: Number, required: true, min: 0 }
  },
  { _id: false }
);

const parentOrderSchema = new mongoose.Schema(
  {
    orderGroupId: { type: String, required: true, unique: true, index: true },
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true, index: true },
    childOrderIds: [{ type: String, required: true }],
    childSummaries: { type: [childSummarySchema], default: [] },
    aggregateOrderStatus: { type: String, required: true, index: true },
    aggregatePaymentStatus: { type: String, required: true, index: true },
    paymentMethod: { type: String, enum: ['COD', 'ONLINE'], required: true },
    shippingAddress: {
      label: { type: String, default: 'Home' },
      fullName: { type: String, required: true },
      phone: { type: String, required: true },
      line1: { type: String, required: true },
      line2: { type: String, default: '' },
      city: { type: String, required: true },
      state: { type: String, required: true },
      pincode: { type: String, required: true }
    },
    deliverySlot: {
      date: { type: Date, required: true },
      timeWindow: { type: String, required: true }
    },
    deliverySlotReleased: { type: Boolean, default: false },
    totalMRP: { type: Number, required: true, min: 0 },
    totalDiscount: { type: Number, required: true, min: 0 },
    couponCode: { type: String, default: '' },
    couponDiscount: { type: Number, default: 0, min: 0 },
    deliveryFee: { type: Number, required: true, min: 0 },
    tax: { type: Number, required: true, min: 0 },
    totalAmount: { type: Number, required: true, min: 0 }
  },
  { timestamps: true }
);

parentOrderSchema.index({ userId: 1, createdAt: -1 });
parentOrderSchema.index({ userId: 1, aggregateOrderStatus: 1, createdAt: -1 });

module.exports = mongoose.model('ParentOrder', parentOrderSchema);

File: spring_back/src/main/java/com/groceryapp/model/ParentOrder.java
package com.groceryapp.model;

import jakarta.persistence.Column;
import jakarta.persistence.CollectionTable;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Embeddable;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "parent_orders")
public class ParentOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    @Column(unique = true)
    private String orderGroupId;
    private String userId;
    @ElementCollection
    @CollectionTable(name = "parent_order_child_order_ids", joinColumns = @JoinColumn(name = "parent_order_id"))
    @Builder.Default
    private List<String> childOrderIds = new ArrayList<>();
    @ElementCollection
    @CollectionTable(name = "parent_order_child_summaries", joinColumns = @JoinColumn(name = "parent_order_id"))
    @Builder.Default
    private List<ChildSummary> childSummaries = new ArrayList<>();
    private String aggregateOrderStatus;
    private String aggregatePaymentStatus;
    private String paymentMethod;
    @Embedded
    private Order.Address shippingAddress;
    @Embedded
    private Order.DeliverySlotInfo deliverySlot;
    private Boolean deliverySlotReleased;
    private Double totalMRP;
    private Double totalDiscount;
    private String couponCode;
    private Double couponDiscount;
    private Double deliveryFee;
    private Double tax;
    private Double totalAmount;
    private Instant createdAt;
    private Instant updatedAt;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    @Embeddable
    public static class ChildSummary {
        private String orderId;
        private String sellerId;
        private String orderStatus;
        private String paymentStatus;
        private Double totalAmount;
    }
}
